### ISSUE TITLE
GaN example + DynaPhoPy

### ISSUE THREAD

**Issue by antonf-ekb** [abelcarreras/phonolammps#10]
Dear developers!
First of all, I would like to say that PhonoLammps seems very interesting for me, especially combined with DynaPhoPy. However, when I try to run a **finite temperature** calculation for GaN (based on the input from /examples/GaN) as is shown in Pd/ Si examples, I obtain a phonon bandstructure containing a lof of imaginary frequencies. The same problem I face when I try to calculate the phonon spectrum at finite temperature for a compound I am interested in. Is it a common problem with systems containing different elements? 
Here is the [code](https://drive.google.com/file/d/1JpTFAiZUK52x_V13M5KpqNtCV9M07Sot/view?usp=share_link) I used to obtain the results. 
Best regards,
Anton.

**Comment by abelcarreras** [comment]
I noticed that you are generating a MD supercell of 4x4x4 for dynaphopy and using the force constants calculated with a 3x3x3 supercell. This means that the commensurate points in the supercells used in MD and force constants are not the same and you may get some issues from this. Also GaN crystal has less symmetry than Si crystal and I expect to require longer MD to get similar quality results. 

I recommend to use a supercell for the MD that is proportional to the 3x3x3 supercell used for force constants. Also start with a very low temperature to check that matches (is close to) the harmonic model and then test increasing temperatures little by little.

**Comment by abelcarreras** [comment]
If you still get issues then I will try to test by myself.

**Comment by antonf-ekb** [comment]
Thanks for the reply! I've tried to make the same sizes of the supercells, decrease the temperature down to 10K and increased the time of simulation 5 times, but unfortunatelly the problem is still there.

**Comment by antonf-ekb** [comment]
OK, looks like I did something wrong before bacause now it indeed gives good results...However, I still can't get rid of imaginary frequencies for a compound I'm interested in. I use a potential fitted by myself, which reproduces reference DFT-data, I've performed some finite temperature LAMMPS simulations with it as well.. I guess to obtain more accurate simulation, one should increase relaxation time/supercell size, but here I have another issue. I noticed that generate_lammps_trajectory() runs at a single core in contrast to calculation.get_renormalized_phonon_dispersion_bands(), which leads to that lammps simulation runs very slowly. Should I install LAMMPS with OpenMP support to make generate_lammps_trajectory() run in parallel?

**Comment by abelcarreras** [comment]
Yes, basically dynaphopy makes use of LAMMPS python interface to obtain the MD trajectory.  The technical details about parallelization, potential, etc.. very much depend on your LAMMPS compilation. If you need it more computational power,  it is also possible to run the MD separately (maybe in a cluster computer), store the trajectory in a lammpstrj file,  and then use dynaphopy in usual way with this file.

**Comment by antonf-ekb** [comment]
Dear Abel,
If I may ask you one more question. You say that it is possible to run the MD separately and then read a lammpstrj file. In the examples at DynaPhoPy repository I've found how to read the trajectory file, but I wonder how to write an .in file for LAMMPS simulation with the same parameters as set by generate_lammps_trajectory() function?

**Comment by abelcarreras** [comment]
You can check dynaphopy manual's example:

https://abelcarreras.github.io/DynaPhoPy/examples.html

and compare with the LAMMPS input you are using in phonolammps. There are a few extra lines to control de NVT MD. Also you can check the log.lammps file generated by phonolammps to get extra details. Finally you can check the **generate_lammps_trajectory** function itself: 

https://github.com/abelcarreras/DynaPhoPy/blob/70ae51aef695f1bac9f01d4d7f6bb5e3e63ad65b/dynaphopy/interface/lammps_link.py#L8

**Comment by antonf-ekb** [comment]
Thank you for the hints! Do I understand correctly that [this example](https://abelcarreras.github.io/DynaPhoPy/examples.html) does not contain relaxation as provided by the generate_lammps_trajectory() function, and to add it we have to add `run N_relax` before `dump` command, where N_relax=relaxation_time/time_step?

**Comment by abelcarreras** [comment]
In this example the final call to dynaphopy:

`$ dynaphopy input_file trajectory.lammpstrj -n 500000 -ts 0.001`

uses **-n 500000**, which means that it takes the last 500000 steps of the simulation. Since the LAMMPS simulation has 1000000 steps total it means that discards the first 500000 for the dynaphopy calculation. 
However, it is true that you can specify the relaxation steps directly in LAMMPS input as you suggest and then ignore **-n** option. Both strategies are OK.

**Comment by antonf-ekb** [comment]
I've noticed a strange thing when I run a LAMMPS calculation separately and then read the trajectory via read_lammps_trajectory() function. Everything looks fine if the supercell for LAMMPS input is generated manually (using Dynaphopy) but breaks (imaginary modes appear) if the supercell is generated via `replicate n n n` command in the LAMMMS input file. It is interesting, why this can happen.
